common/flatpak-dir.c

```diff
@@ -392,6 +390,39 @@ flatpak_remote_state_lookup_sparse_cache (FlatpakRemoteState *self,
   return FALSE;
 }
 
+static gboolean
+flatpak_remote_state_save_summary (FlatpakRemoteState *self,
+                                   GFile              *dir,
+                                   GCancellable       *cancellable,
+                                   GError             **error)
+{
+  g_autoptr(GFile) summary_file = g_file_get_child (dir, "summary");
+  g_autoptr(GBytes) summary_bytes = NULL;
+
+  /* For non-p2p case we always require a summary */
+  if (!flatpak_remote_state_ensure_summary (self, error))
```

Comment:
This doesn’t check for the P2P/non-P2P case though.

Reply:
Maybe add `g_assert (self->collection_id == NULL)`?

Reply:
Well, this code is only used in the non-p2p case which is why we have the ensure_summary() call, which essentially doubles as your assert.